src/components/Navigation.jsx

```diff
@@ -1,19 +1,55 @@
-import React, { useState } from "react";
-import { Link, useLocation } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { Link, useLocation, useNavigate } from "react-router-dom";
 import { Bars3Icon, XMarkIcon } from "@heroicons/react/24/outline";
 
 function Navigation() {
   const location = useLocation();
+  const navigate = useNavigate();
   const [menuOpen, setMenuOpen] = useState(false);
+  const [userData, setUserData] = useState(null);
+  const token = localStorage.getItem("token");
+
+  const handleLogout = () => {
+    localStorage.removeItem("token");
+    setUserData(null); 
+    setMenuOpen(false);
+    navigate("/login"); 
+  };
+
+  useEffect(() => {
+    setMenuOpen(false);
+  }, [location.pathname]);
+
+  useEffect(() => {
+
+    {/* 
+      Ask Akhil about this
+
+      if (location.pathname.startsWith("/login") && token) {
+        navigate("/account");
+      }
+    */}
+
+    if (token) {
+      fetch("https://gg-pac-team7-back.onrender.com/api/v1/profile", {
```

Comment:
Switch this to use the `.env` variable for the backend URL: `import.meta.env.VITE_REACT_APP_BACKEND_URL`, which will have everything before /api

Reply:
Done!